.travis.yml

```diff
@@ -103,21 +113,30 @@ matrix:
             - *linux_deps
 
 before_install:
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then
+      sudo apt-get update;
+      sudo apt-get install -y python3 python3-venv python3-pip python3-setuptools;      
+    fi
+  - python3 -m pip install --user virtualenv
+  - python3 -m venv buildenv
+  - source buildenv/bin/activate
+  - python3 -m pip install --upgrade pip
+  - pip install scons
+  - scons --version
   - eval "${MATRIX_EVAL}"
   - if [ "$STATIC_CHECKS" = "yes" ]; then
       unset SCONS_CACHE;
     fi
 
 install:
-  - pip install --user scons;
   - if [ "$TRAVIS_OS_NAME" = "linux" ] && [ "$PLATFORM" = "android" ]; then
       export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64;
       export PATH=/usr/lib/jvm/java-8-openjdk-amd64/jre/bin:${PATH};
       java -version;
       misc/travis/android-tools-linux.sh;
     fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then
-      export PATH=${PATH}:/Users/travis/Library/Python/2.7/bin;
+  - if [ "$TRAVIS_OS_NAME" = "osx" ] && [ "$PLATFORM" = "android" ]; then
+      misc/travis/android-tools-osx.sh;
```

Comment:
We're no longer building Android on OSX, and this script has been removed a while ago.

Reply:
okay will rebase and remove